### PR TITLE
feat: Updated LangChain example

### LINKED ISSUE #171: (langchain) Create example application
Create an example application utilizing LangChain. It should include the following:

+ Utilize OpenAI as the underlying LLM in order to keep things easy to follow
+ Chaining is not necessary, but a callout to the [LangChain documentation](https://js.langchain.com/docs/expression_language/get_started) would be helpful
+ Require the user to provide their own OpenAI API key, New Relic application name, and New Relic agent key
+ Endpoint that creates completions without streaming
+ Endpoint that creates completions with streaming
+ Endpoint that creates a chat completion, stores relevant ids from `getLlmMessageIds`, and in a separate API call record a feedback event with said ids

You can follow https://github.com/newrelic/newrelic-node-examples/tree/782f522f2f3ee3a012f89897cfaa0deec6e54232/openai as an example.

### PR DESCRIPTION
**Pull request by mrickard**
## Proposed Release Notes
Updated LangChain example to show vector search and tools calls.

## Links
Closes #171 
Closes NR-218735

### CODE REVIEW

File: langchain/custom-tool.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const { Tool } = require('@langchain/core/tools')
+const data = {
```
Comment (bizob2828) [inline langchain/custom-tool.js:8]: The response must be a scalar. so you should remove the inner object and just have the value of response be the value of the parent key

**Review by bizob2828 — CHANGES_REQUESTED** [review]
The chat completions aren't loading any langchain instrumentation because you aren't creating a chain. You are just calling openai related interactions. Also the sample langchain messages aren't an idiomatic langchain call.  You typically provide it a template.  I would check out our versioned tests.

File: langchain/custom-tool.js (hunk 1 above)
Comment (mrickard) [reply langchain/custom-tool.js:8]: OK. That's updated.

File: langchain/index.js [hunk 2]
```diff
@@ -113,7 +112,7 @@ fastify.post('/feedback', (request, reply) => {
 
 fastify.post('/tools', async (request, reply) => {
   const {
-    message = 'Say this is a test.',
+    topic = 'Say this is a test.',
     temperature = 0.5
```
[the remark this answers is not in the document]
Comment (mrickard) [reply langchain/index.js:116]: I'll remove the temperature from this one. It's already removed from the example requests in the README.

**Pull request merged**